add AppSyncIdentity

daviskoh opened the pull request:
*Issue #, if available:*

*Description of changes:*

Adds new `AppSyncIdentity` struct which represents the `.identity` field of [context](https://docs.aws.amazon.com/appsync/latest/devguide/resolver-context-reference.html). Other context fields are left out because they are all custom structures that depend on user implementation.

One thing to note is that `AppSyncIdentity.Claims` is of type `map[string]interface{}` which sucks, but Claims values can be either `Number` or `String` json types and there may be custom claims added by the user...

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

bmoffatt commented on events/appsync.go:
```diff
@@ -9,6 +9,20 @@ type AppSyncResolverTemplate struct {
 	Payload   json.RawMessage  `json:"payload"`
 }
 
+// AppSyncIdentity contains information about the caller. The shape of this section depends on the authorization type of your AWS AppSync API
```
Seems that there'd need to be 2 types added, rather than the union of both shapes. https://docs.aws.amazon.com/appsync/latest/devguide/resolver-context-reference.html
* `AppSyncIAMIdentity`
* `AppSyncCognitoIdentity`

daviskoh replied:
Unmarshalling / marshalling `$context.identity` into the single type works either way, but I'll separate out the type if that's what's desired.

daviskoh commented:
hmmm also noticing now the test are off...will ping you when resolved.

daviskoh commented:
@bmoffatt i've updated everything:
- separated out AppSyncIdentity into separate types for IAM & Cognito
- fixed tests to actually check marshalling / unmarshalling of identity json

u mind having a look? 😃
